### PR TITLE
Enable Ad Serving Version 2 for 50% of Nightly users

### PR DESCRIPTION
**Pull request by moritzhaller**

### CODE REVIEW

**Review by tmancey — APPROVED** [review]
LGTM

**Comment by btlechowski** [comment]
Good to merge. Basic tests done for v1 and v2.

Verification passed on

Brave | 1.34.23 Chromium: 96.0.4664.45 (Official Build) nightly (64-bit)
-- | --
Revision | 76e4c1bb2ab4671b8beba3444e61c0f17584b2fc-refs/branch-heads/4664@{#947}
OS | Ubuntu 18.04 LTS

### Ad serving v2

Verified Ad serving v2 was used in griffin
image

Verified notification ad was shown:
```
[4636:4636:1119/043713.284061:VERBOSE1:eligible_ad_notifications_v2.cc(38)] Get eligible ad notifications:
[4636:4636:1119/043713.288214:VERBOSE1:ad_notification_serving.cc(144)] Found 1 eligible ads
[4636:4636:1119/043713.288385:VERBOSE1:ad_notification_serving.cc(263)] Serving ad notification:
  uuid: f0cf9e8f-bf8f-4f18-82f6-9338a3f0e7ff
  creativeInstanceId: 7898a259-664a-4052-8382-9c7279ae3249
  creativeSetId: 5e96fecc-2926-4c72-bb76-4e745bf6b539
  campaignId: 0d01425b-c970-4a25-b2d0-ccf748cfaf6e
  advertiserId: ecbcc833-8b1d-4867-98f7-ad2341396ce8
  segment: untargeted
  title: Ad Set 2 - Creative 1 - Ad 1
  body: Ad Set 2 - Creative 1 - Ad 1
  targetUrl: https://youtube.com
```

User was paid for it
```
[4636:4636:1119/043714.337592:VERBOSE1:confirmations.cc(283)] Added 1 unblinded payment token with an estimated redemption value of 0.05 BAT, you now have 1 unblinded payment tokens

```

Verified inline ad:
```
[5188:5188:1119/044550.112458:VERBOSE1:eligible_inline_content_ads_v2.cc(39)] Get eligible inline content ads:
[5188:5188:1119/044550.117239:VERBOSE1:inline_content_ad_serving.cc(86)] Found 1 eligible ads
[5188:5188:1119/044550.117474:VERBOSE1:inline_content_ad_serving.cc(129)] Serving inline content ad:
  uuid: b85ef23f-0882-4133-a184-bed9a7055cb9
  creativeInstanceId: e4958d00-e35c-4134-a408-1fbcf274d5ae
  creativeSetId: 8b848823-c47f-43de-a753-2efa8dea6863
  campaignId: 07de7f05-ab32-4af0-a350-a1bce4427eac
  advertiserId: 93130af3-2def-4ecb-b836-b3772e73b3c9
  segment: untargeted
  title: Staging - Inline Test 1
  description: Fung Brave News Inline Display Ad - mac/linux
  imageUrl: https://pcdn.brave.com/brave-ads/468beb93-2458-5d3e-8d7b-bc1f387d2e9b.webp.pad
  dimensions: 900x750
  ctaText: Work at Brave
  targetUrl: https://brave.com/careers
```

Verified user is paid for it:
```
[5188:5188:1119/044558.061065:VERBOSE1:confirmations.cc(283)] Added 1 unblinded payment token with an estimated redemption value of 0.01 BAT, you now have 1 unblinded payment tokens
```

### Ad serving v1

image


Verified notification ad:
```
[7812:7812:1119/045252.501653:VERBOSE1:eligible_ad_notifications_v1.cc(39)] Get eligible ad notifications:
[7812:7812:1119/045252.502652:VERBOSE1:eligible_ad_notifications_v1.cc(140)] Get eligible ads for untargeted segment
[7812:7812:1119/045252.504470:VERBOSE2:exclusion_rules_base.cc(123)] creativeSetId c7e7a381-4c74-4dd9-9b6d-4edf9330454b excluded as not associated with an advertiser split test group
[7812:7812:1119/045252.505414:VERBOSE2:ad_priority.h(32)] 13 ads with a priority of 1 in bucket 1
[7812:7812:1119/045252.505796:VERBOSE1:ad_notification_serving.cc(144)] Found 13 eligible ads
[7812:7812:1119/045252.505825:VERBOSE1:ad_notification_serving.cc(263)] Serving ad notification:
  uuid: d39b5cad-bcda-4427-b7ee-a73481fa9c62
  creativeInstanceId: 7898a259-664a-4052-8382-9c7279ae3249
  creativeSetId: 5e96fecc-2926-4c72-bb76-4e745bf6b539
  campaignId: 0d01425b-c970-4a25-b2d0-ccf748cfaf6e
  advertiserId: ecbcc833-8b1d-4867-98f7-ad2341396ce8
  segment: untargeted
  title: Ad Set 2 - Creative 1 - Ad 1
  body: Ad Set 2 - Creative 1 - Ad 1
  targetUrl: https://youtube.com
```

Verified user was paid:
```
[7812:7812:1119/045253.780843:VERBOSE1:confirmations.cc(283)] Added 1 unblinded payment token with an estimated redemption value of 0.05 BAT, you now have 1 unblinded payment tokens
```


Verified inline ad
```
[8218:8218:1119/045829.064927:VERBOSE1:eligible_inline_content_ads_v1.cc(40)] Get eligible inline content ads:
[8218:8218:1119/045829.065915:VERBOSE1:eligible_inline_content_ads_v1.cc(150)] Get eligible ads for untargeted segment
[8218:8218:1119/045829.067378:VERBOSE2:ad_priority.h(32)] 2 ads with a priority of 1 in bucket 1
[8218:8218:1119/045829.067554:VERBOSE1:inline_content_ad_serving.cc(86)] Found 2 eligible ads
[8218:8218:1119/045829.067700:VERBOSE1:inline_content_ad_serving.cc(129)] Serving inline content ad:
  uuid: c24aa38e-267c-4afd-a40d-2b59ea80f813
  creativeInstanceId: 033a95ca-963b-460c-b203-8181346bd41c
  creativeSetId: 6c875e23-00ea-4f1b-805b-5d52559f96da
  campaignId: 8a5ca6f0-f919-461b-8bf2-4848e34ab616
  advertiserId: aa1d1cdb-ec45-42d4-86d0-ea353e1babd4
  segment: untargeted
  title: Makita LF1000 Flip Over Saw
  description: Makita
  imageUrl: https://pcdn.bravesoftware.com/brave-ads/image-1.jpg.pad
  dimensions: 900x750
  ctaText: Learn more
  targetUrl: https://www.powertools2u.co.uk/makita-lf1000-flip-over-sawo
```

Verified user is paid:
```
[8218:8218:1119/045838.913898:VERBOSE1:confirmations.cc(283)] Added 1 unblinded payment token with an estimated redemption value of 0.01 BAT, you now have 2 unblinded payment tokens
```

**Review by kjozwiak — COMMENTED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-variations/pull/160#issuecomment-974272028.

**Review by kjozwiak — APPROVED** [review]

**Pull request merged**
